docs/hive-integration/kube-api-getting-started.md

```diff
@@ -0,0 +1,493 @@
+time="2021-06-28T21:01:31Z" level=info msg="Updated clusterDeployment demo-worker4/single-node" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).updateIfNeeded" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:714" agent_cluster_install=test-agent-cluster-install agent_cluster_install_namespace=demo-worker4 cluster_deployment=single-node cluster_deployment_namespace=demo-worker4 go-id=684 request_id=98cac786-f3c2-4237-ae7c-f5c1a9e6e85b
+time="2021-06-28T21:01:31Z" level=info msg="ClusterDeployment Reconcile ended" func="github.com/openshift/assisted-service/internal/controller/controllers.(*ClusterDeploymentsReconciler).Reconcile.func1" file="/go/src/github.com/openshift/origin/internal/controller/controllers/clusterdeployments_controller.go:112" agent_cluster_install=test-agent-cluster-install agent_cluster_install_namespace=demo-worker4 cluster_deployment=single-node cluster_deployment_namespace=demo-worker4 go-id=684 request_id=98cac786-f3c2-4237-ae7c-f5c1a9e6e85b
+```
+
+### Generate Cluster Discovery Image
+
+#### 1. Optional - Create [NMStateConfig](README.md#NMStateConfig)
+* Needed for optional network configuration, such as static IP addresses and more.
+
+```yaml
+cat <<EOF | kubectl create -f -
+apiVersion: agent-install.openshift.io/v1beta1
+kind: NMStateConfig
+metadata:
+  name: mynmstateconfig
+  namespace: demo-worker4
+  labels:
+    demo-nmstate-label: some-value # both the label name and value must be included in InfraEnv nmStateConfigLabelSelector matchLabels
+spec:
+  config:
+    interfaces:
+      - name: eth0
+        type: ethernet
+        state: up
+        mac-address: 02:00:00:80:12:14
+        ipv4:
+          enabled: true
+          address:
+            - ip: 192.168.111.30
+              prefix-length: 24
+          dhcp: false
+      - name: eth1
+        type: ethernet
+        state: up
+        mac-address: 02:00:00:80:12:15
+        ipv4:
+          enabled: true
+          address:
+            - ip: 192.168.140.30
+              prefix-length: 24
+          dhcp: false
+    dns-resolver:
+      config:
+        server:
+          - 192.168.126.1
+    routes:
+      config:
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.168.111.1
+          next-hop-interface: eth1
+          table-id: 254
+        - destination: 0.0.0.0/0
+          next-hop-address: 192.168.140.1
+          next-hop-interface: eth1
+          table-id: 254
+  interfaces:
+    - name: "eth0"
+      macAddress: "02:00:00:80:12:14"
+    - name: "eth1"
+      macAddress: "02:00:00:80:12:15"
+EOF
+```
+
+#### 2. Create [InfraEnv](README.md#InfraEnv)
```

Comment:
It would be useful to add a sentence here about what this is. Maybe "The InraEnv describes everything about the infrastructure environment that is needed in order to create a discovery ISO."